explore/src/main/scala/explore/config/SpectroscopyModesTable.scala

```diff
@@ -418,8 +424,16 @@ object SpectroscopyModesTable {
                                     itcResults.async,
                                     itcProgress.async
                       )
-                }.orEmpty
+                  }
+                  .getOrElse(IO.unit)
               )
+
+          val isCacheEmpty = itcResults.value.get.isEmpty
+
+          // Only send the requests if the cache is empty, send the visible rows first
+          submitRows(range.value.foldMap(visibleRows(_, sortedRows)) ++ sortedRows)
+            .whenA(isCacheEmpty)
```

Comment:
What happens if any of the ITC parameters change but the cache is not emptied?

Reply:
You're right, this is failing